plugin-repos/ufc-scoreboard/base_odds_manager.py

```
@@ -0,0 +1,284 @@
+                odds_data = self.get_odds(
+                    sport, league, event_id, comp_id, update_interval_seconds
+                )
+                if odds_data:
+                    results[comp_id] = odds_data
+            except Exception as e:
+                self.logger.error(f"Error fetching odds for event {event_id}/{comp_id}: {e}")
+                continue
+
+        return results
+
+    def clear_cache(self, sport: str = None, league: str = None, event_id: str = None):
```

Ruff (0.15.0)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)